### PR TITLE
[NCL-7649] Add SCM external link button

### PR DESCRIPTION
**Pull request by DnsZhou**
Rebased from https://github.com/project-ncl/pnc-web-ui-react/pull/185 due to too many conflicts

### CODE REVIEW

File: src/components/ScmRepositoryUrl/ScmRepositoryUrl.tsx [hunk 1]
```diff
@@ -1,54 +1,63 @@
...
-  return (
-    <CopyToClipboard isInline={isInline} suffixComponent={showGerritButton && <GerritButton isInline={isInline} url={url} />}>
-      {url}
+export const ScmRepositoryUrl = ({ internalScmRepository, externalScmRepository, isInline }: IScmRepositoryUrlProps) => {
+  const originalUrl = internalScmRepository ? internalScmRepository.internalUrl : externalScmRepository!.externalUrl;
+  const parsedUrl = !originalUrl
+    ? null
+    : internalScmRepository
+    ? parseInternalScmRepositoryUrl({ url: originalUrl })
+    : parseExternalScmRepositoryUrl({ url: originalUrl });
```
Comment (patrikk0123) [inline src/components/ScmRepositoryUrl/ScmRepositoryUrl.tsx:56]: What about:

```
interface IScmRepositoryUrlProps {
  scmRepository: SCMRepository;
 variant: 'internal' | 'external';
 isInline?: boolean;
}

export const ScmRepositoryUrl = ({ scmRepository, variant, isInline }: IScmRepositoryUrlProps) => {
    const originalUrl = variant === 'internal'
        ? scmRepository.internalUrl
        : scmRepository.externalUrl
    const parsedUrl = variant === 'internal'
        ? parseInternalScmRepositoryUrl({ url: originalUrl })
        : parseExternalScmRepositoryUrl({ url: originalUrl });
```

File: src/utils/utils.ts [hunk 2]
```diff
@@ -35,19 +35,47 @@ export const transformateDateFormat = (date: Date) => {
   return `${year}-${monthString}-${dayString} ${hourString}:${minuteString}`;
 };
 
-interface IParseInternalRepositoryUrl {
-  internalUrl: string;
+interface IParsedScmRepositoryUrl {
```
Comment (patrikk0123) [inline src/utils/utils.ts:38]: `IScmRepositoryUrl` -- this one is not yet parsed.

File: src/utils/utils.ts [hunk 3]
```diff
@@ -35,19 +35,47 @@ export const transformateDateFormat = (date: Date) => {
...
+export interface IParsedUrl {
+  url: string;
+  displayName: string;
+}
+
+/**
+ * Parses internal SCM Repository URL to Gerrit gitweb link of the SCM Repository.
+ *
+ * @param internalUrl - The internalUrl to be parsed
+ * @returns parsedUrl contains parsedUrl and displayName
```
Comment (patrikk0123) [inline src/utils/utils.ts:51]: `Object containing parsed URL and display name representing URL`

File: src/utils/utils.ts [hunk 4]
```diff
@@ -35,19 +35,47 @@ export const transformateDateFormat = (date: Date) => {
...
+};
+
 /**
- * Parses internal repository url to Gerrit gitweb link of the project.
+ * ParsesSCM Repository URL to gitweb link of the SCM Repository.
  *
- * @param scmRepository - SCM Repository containing internalUrl field
- * @returns SCM Repository name
+ * @param externalUrl - The externalUrl to be parsed
+ * @returns  Object contains url and the base of the external url
```
Comment (patrikk0123) [inline src/utils/utils.ts:64]: `Object containing parsed URL and display name representing URL`

File: src/utils/utils.ts [hunk 5]
```diff
@@ -35,19 +35,47 @@ export const transformateDateFormat = (date: Date) => {
...
+  return { url: 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary', displayName: 'Gerrit' };
+};
+
 /**
- * Parses internal repository url to Gerrit gitweb link of the project.
+ * ParsesSCM Repository URL to gitweb link of the SCM Repository.
  *
- * @param scmRepository - SCM Repository containing internalUrl field
- * @returns SCM Repository name
+ * @param externalUrl - The externalUrl to be parsed
```
Comment (patrikk0123) [inline src/utils/utils.ts:63]: `url`

File: src/utils/utils.ts [hunk 6]
```diff
@@ -35,19 +35,47 @@ export const transformateDateFormat = (date: Date) => {
...
 
+export interface IParsedUrl {
+  url: string;
+  displayName: string;
+}
+
+/**
+ * Parses internal SCM Repository URL to Gerrit gitweb link of the SCM Repository.
+ *
+ * @param internalUrl - The internalUrl to be parsed
```
Comment (patrikk0123) [inline src/utils/utils.ts:50]: `url`

File: src/components/ScmRepositoryUrl/ScmRepositoryUrl.tsx (hunk 1 above)
Comment (DnsZhou) [reply src/components/ScmRepositoryUrl/ScmRepositoryUrl.tsx:56]: using different repository objects was suggested by Martin, @matedo1 wdyt?

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Review by matedo1 — CHANGES_REQUESTED** [review]
Some last small changes

**Comment by DnsZhou** [comment]
@matedo1  ready for review again

File: src/utils/urlParseHelper.ts [hunk 7]
```diff
@@ -3,42 +3,47 @@ interface IScmRepositoryUrl {
...
+  return { scmRepositoryUrl: url, webUrl: 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary', name: 'Gerrit' };
 };
 
 /**
  * ParsesSCM Repository URL to gitweb link of the SCM Repository.
  *
  * @param url - The externalUrl to be parsed
- * @returns  Object containing parsed URL and display name representing URL
+ * @returns  1) null when url was not successfully parsed, otherwise
+ * 2) object containing parsed URL and display name representing URL
```
Comment (matedo1) [inline src/utils/urlParseHelper.ts:29]: `Object containing scmRepository URL, parsed URL and display name representing URL`

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Comment by DnsZhou** [comment]
@matedo1  ready for review, Thanks

**Review by matedo1 — APPROVED** [review]

**Pull request merged**